### PR TITLE
fix repair queue

### PR DESCRIPTION
**Pull request by littleskunk**
What: Remove segments from the queue that doesn't need to be repaired.

Why: If the repair job returns an error the segment will stay on the queue.
```
2019-07-01T11:47:36.762+0200    ESC[31mERRORESC[0m      repairer/repairer.go:132        repair failed:  {"error": "repairer error: repair failed: segment error: segment 1e77f236-1749-4376-948a-ef133e4c57c4/l/transfersh/mTjr6QygUUr4ADp3mssF5_JyUrU3VYWwA9Yh4jNmnqw9/hyzuJlPgHAqXFwEz4l85M9KD6UiopmSI2972RHdfbpTR81E with 83 pieces above repair threshold 35", "errorVerbose": "repairer error: repair failed: segment error: segment 1e77f236-1749-4376-948a-ef133e4c57c4/l/transfersh/mTjr6QygUUr4ADp3mssF5_JyUrU3VYWwA9Yh4jNmnqw9/hyzuJlPgHAqXFwEz4l85M9KD6UiopmSI2972RHdfbpTR81E with 83 pieces above repair threshold 35\n\tstorj.io/storj/pkg/datarepair/repairer.(*Service).worker:143\n\tstorj.io/storj/pkg/datarepair/repairer.(*Service).process.func1:130\n\tstorj.io/storj/internal/sync2.(*Limiter).Go.func1:41"}
storj.io/storj/pkg/datarepair/repairer.(*Service).process.func1
        /home/storj/source/pkg/datarepair/repairer/repairer.go:132
storj.io/storj/internal/sync2.(*Limiter).Go.func1
        /home/storj/source/internal/sync2/limiter.go:41

2019-07-01T12:51:50.130+0200    ESC[31mERRORESC[0m      repairer/repairer.go:132        repair failed:  {"error": "repairer error: repair failed: segment error: segment 1e77f236-1749-4376-948a-ef133e4c57c4/l/transfersh/mTjr6QygUUr4ADp3mssF5_JyUrU3VYWwA9Yh4jNmnqw9/hyzuJlPgHAqXFwEz4l85M9KD6UiopmSI2972RHdfbpTR81E with 85 pieces above repair threshold 35", "errorVerbose": "repairer error: repair failed: segment error: segment 1e77f236-1749-4376-948a-ef133e4c57c4/l/transfersh/mTjr6QygUUr4ADp3mssF5_JyUrU3VYWwA9Yh4jNmnqw9/hyzuJlPgHAqXFwEz4l85M9KD6UiopmSI2972RHdfbpTR81E with 85 pieces above repair threshold 35\n\tstorj.io/storj/pkg/datarepair/repairer.(*Service).worker:143\n\tstorj.io/storj/pkg/datarepair/repairer.(*Service).process.func1:130\n\tstorj.io/storj/internal/sync2.(*Limiter).Go.func1:41"}
```

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: pkg/storage/segments/repairer.go [hunk 1]
```diff
@@ -85,7 +85,7 @@ func (repairer *Repairer) Repair(ctx context.Context, path storj.Path) (err erro
 	// repair not needed
 	if int32(numHealthy) > pointer.Remote.Redundancy.RepairThreshold {
 		mon.Meter("repair_unnecessary").Mark(1)
-		return Error.New("segment %v with %d pieces above repair threshold %d", path, numHealthy, pointer.Remote.Redundancy.RepairThreshold)
+		return nil
```
Comment (stefanbenten) [inline pkg/storage/segments/repairer.go]: Lets add a log line about that, but then LGTM

Comment (littleskunk) [reply pkg/storage/segments/repairer.go]: unfortunately there is no logger available :/

Comment (mobyvb) [reply pkg/storage/segments/repairer.go]: it would be easy to make one accessible as part of this PR
https://github.com/storj/storj/pull/2365/files

Comment (stefanbenten) [reply pkg/storage/segments/repairer.go]: Can you take care of that once this is merged? @mobyvb

Comment (mobyvb) [reply pkg/storage/segments/repairer.go]: Yeah, I'll add it to the PR. Should only be a few lines

**Pull request merged**